go-controller/pkg/ovn/master.go

```diff
@@ -764,6 +766,50 @@ func (oc *Controller) ensureNodeLogicalNetwork(nodeName string, hostSubnets []*n
 		return err
 	}
 
+	nodePrimaryIP, err := util.GetNodeLocalIP(node)
+	if err != nil || nodePrimaryIP == nil {
+		return fmt.Errorf("Couldn't obtain primary IP for node %q: %v", node.Name, err)
+	}
+
+	hostNetworkPolicyIPs = append(hostNetworkPolicyIPs, *nodePrimaryIP)
```

Comment:
why are you adding the host primary IP here? If the traffic comes through mp0 it will be snat'ed and if it comes via GR it will be snat'ed.

Reply:
when it truly comes from the node, won't it use the node IP? do we SNAT in that case to mp0?

Reply:
Yeah it will get snat to mp0. I think its safe to remove.